adam/perception/perception_graph.py

```diff
@@ -519,11 +560,17 @@ class PerceptionGraphPatternMatching:
         validator=instance_of(PerceptionGraph)
     )
 
+    # Callable object for debugging purposes. We use this to track the number of calls to match and render the graphs.
+    rendering_debug_function: "DebugRenderingObject" = attrib(
```

Comment:
* Call this `debug_callback`
* Make the type `Optional[Callable[[DiGraph, Dict[Any, Any]]]]`. You probably want to make a type alias for the `Callable` portion to make it more readable.
* Make it default to `None` (we only want to do the debug rendering on request)